### ISSUE TITLE
Android functional tests

### ISSUE THREAD

**Issue by vmrob** [ccbrown/needy#151]
I think they're needed and an easy way to start this would be to utilize a containerized android build environment.

**Comment by ccbrown** [comment]
Agreed. One thing in particular: Testing the various runtime / api level combinations would be a really good thing.
